ui/src/components/pipeline/Pipeline.tsx

```diff
@@ -83,23 +85,72 @@ export function Pipeline() {
           )
             .then((response) => response.json())
             .then((json) => {
-              const processinRates = {} as ProcessingRates;
-              processinRates.ratePerMin =
+              const vertexMetrics = {} as VertexMetrics;
+              vertexMetrics.ratePerMin =
                 json["processingRates"]["1m"].toFixed(2);
-              processinRates.ratePerFiveMin =
+              vertexMetrics.ratePerFiveMin =
                 json["processingRates"]["5m"].toFixed(2);
-              processinRates.ratePerFifteenMin =
+              vertexMetrics.ratePerFifteenMin =
                 json["processingRates"]["15m"].toFixed(2);
-              vertexToRateMap.set(vertex.name, processinRates);
+              vertexToMetricsMap.set(vertex.name, vertexMetrics);
             });
         })
-      ).then(() => setVertexRate(vertexToRateMap));
+      ).then(() => setVertexMetrics(vertexToMetricsMap));
     }
+  };
+
+  // This useEffect is used to obtain metrics for a given vertex in a pipeline and refreshes every 5 minutes
+  useEffect(() => {
+    getMetrics();
+    const interval = setInterval(() => {
+      getMetrics();
+    }, 300000);
+
+    return () => clearInterval(interval);
+  }, [pipeline]);
```

Comment:
Add getMetrics as dependency. Don't want to use stale instance once its update to useCallback.